text/0134-remove-span-status.md

```diff
@@ -0,0 +1,89 @@
+field encoded as Google RPC code for recording a result of arbitrary monitored operations from various domains
+(http requests, database queries, file system operations etc). It is hard to define clear and umambigous mapping
+from those domain to a fixed enum originated from RPC.
+
+Therefore the proposal is to remove the `Span.Status` field from Trace API and OTLP.
+
+## Internal details
+
+Several areas are affected by this change and will require a coordinated effort to make this happen.
+
+### Trace API Specification
+
+* Remove all mentions of `Status` and `Set Status` API from Span section.
+* Remove whole `Status` section.
+
+### Trace SDK Specification
+
+No changes needed.
+
+### Trace SDK exporters
+
+Remove translation rules pertaining to `Span.Status`.
+
+### Span semantic conventions
+
+* From HTTP conventions, remove `Status` section.
+* In gRPC conventions, replace the requirement to set span status with the requirement to set span attribute `grpc.status`
+to the canonical status code of the gRPC response.
+
+### OTLP data definition
+
+Rename `Span.status` field to `Span.deprecated_status` field.
+
+### Language API and SDK
+
+* Every language must remove status related code from their implementations of Trace API and SDK.
+* Every language must remove all code in their span exporters which serializes span status.
+* All instrumentation libraries, both manual and automated, must remove all calls to the removed `Span.setStatus` API.
+* All instrumentation libraries, both manual and automated, must adapt new semantic conventions for gRPC spans.
+
+### Collector
+
+Collector will have a transition period during which it has a special handling both for old data,
+which has `Span.Status` present, and new data without Status.
+This will allow for older pipelines, still relying on `Span.Status` to continue to work with as little disruption as possible.
+The exact duration of this transition period will be left for Collector's maintainers to decide.
+
+* If incoming OTLP data has `Span.Status` present, that will NOT be removed and will be passed on as-is.
+* If incoming OTLP data has `Span.Status` present and it is not equal to `OK`,
+Collector will translate that to temporary semantic attributes as follows:
+
+|Status|Tag Key| Tag Value|
+|--|--|--|
+|StatusCanonicalCode | `otel.deprecated_status_code` | Name of the code|
+|Message *(optional)* | `otel.deprecated_status_message` | `{message}`|
+
+* When receiving data in Zipkin format, no special handling will take place.
+All Zipkin tags present will be translated verbatim to OTLP attributes.
+* When exporting data into Zipkin format, no special handling will take place.
+All OTLP attributes will be translated verbatim to Zipkin tags.
+`Span.Status` if present, will NOT be translated.
+* When receiving data in Jaeger format, no special handling will take place.
+Jaeger `error` tag will be translated in the usual way to OTLP `error` attribute.
+See `Open questions` below.
```

Comment:
```suggestion
Jaeger `error` tag will be translated in the usual way to an OTLP error attribute
(see [Open questions](#open-questions)).
```

The use of code formatting was confusing me when reading this, because that made it sound like the attribute actually exists with that name.